backend/server/src/services/userService.ts

```diff
@@ -34,4 +34,16 @@ export class UserService {
 
     await UserModel.create(requestBody);
   }
+
+  public async searchUser(query: string): Promise<User[]> {
+    const searchQuery = new RegExp(query, 'i');
+    return await UserModel.find({
+      $or: [
+        { userName: { $regex: searchQuery } },
+        { email: { $regex: searchQuery } },
+        { firstName: { $regex: searchQuery } },
+        { lastName: { $regex: searchQuery } }
+      ]
+    });
```

Comment:
These should be converted into plain objects before sending them to the client. I think now if you make a request, you will get the document data as well, which is not what we want.